src/aus.h

```diff
@@ -0,0 +1,329 @@
+// License: Apache 2.0. See LICENSE file in root directory.
+// Copyright(c) 2022 Intel Corporation. All Rights Reserved.
+
+#pragma once
+
+#include "types.h"
+
+namespace librealsense
+{
+    
+    class aus_value
+    {
+    public:
+        virtual long get() = 0;
+        virtual void set(long value) = 0;
+        virtual void increment() = 0;
+        virtual void decrement() = 0;
+        virtual void start() = 0;
+        virtual void stop() = 0;
+    };
+
+    class aus_timer : public aus_value {
+    
+    public:
+        aus_timer() : _is_running(false), _total(0), _start(0) { }
+
+        long get()
+        {
+            if (_is_running)
+            {
+                return get_current_time() - _start + _total;
+            }
+            else
+            {
+                return _total;
+            }
+        }
+        
+        void set(long value) {
+            return;
+        }
+
+        void increment()
+        {
+            return;
+        }
+
+        void decrement()
+        {
+            return;
+        }
+
+        void start()
+        {
+            _start = std::chrono::system_clock::to_time_t(
+                std::chrono::system_clock::now());
+            _is_running = true;
+        }
+
+        void stop()
+        {
+            if (!_is_running)
+            {
+                return;
+            }
+            _is_running = false;
+            long current_time = get_current_time();
+            _total += (current_time - _start);
+            _start = current_time;
+        }
+
+
+
+    private:
+        long _start;
+        long _total;
+        bool _is_running;
+
+        time_t get_current_time() 
+        {
+            return std::chrono::system_clock::to_time_t(
+                std::chrono::system_clock::now());
+        }
+
+    };
+
+    class aus_counter : public aus_value
+    {
+    public:
+        aus_counter(long value = 0) : _counter(value), _total(0) { }
+        
+        long get()
+        {
+            return _counter;
+        }
+
+        void set(long value) {
+            _counter = value;
+        }
+
+        void increment()
+        {
+            _counter++;
+        }
+
+        void decrement()
+        {
+            _counter--;
+        }
+
+        void start() { return; }
+
+        void stop() { return; }
+
+
+    private:
+        long _counter;
+        long _total; 
+    };
+
+#ifdef BUILD_AUS
+
+    class aus_data
+    {
+
+    public:
+        aus_data()
+        {
+            _start_time = std::chrono::system_clock::to_time_t(std::chrono::system_clock::now());
+            _librealsense_version = RS2_API_VERSION_STR;
+            _os_name = get_os_name();
+            _platform_name = PLATFORM;
+
+        }
+
+        std::string get_os_name()
```

Comment:
This will be moved to the utilities on a next PR?

Reply:
yes, done in the next PR